### PR TITLE
Testability

### PR DESCRIPTION
**Pull request by bn222**
Use a kind cluster instead of only a control plane for integration tests. This reduces the run time of tests from 42 seconds to 2 seconds on my system for subsequent runs since the cluster is not destroyed.

### CODE REVIEW

File: internal/controller/dpuoperatorconfig_controller_test.go [hunk 1]
```diff
@@ -1,6 +1,9 @@
 package controller
 
 import (
+	// "context"
```
Comment (wizhaoredhat) [inline internal/controller/dpuoperatorconfig_controller_test.go:4]: We should just remove these commented out imports.

File: internal/controller/dpuoperatorconfig_controller_test.go [hunk 2]
```diff
@@ -30,25 +53,14 @@ var (
 	testDpuDaemonName         = "dpu-daemon"
 	testSriovDevicePlugin     = "sriov-device-plugin"
 	testNetworkFunctionNAD    = "dpunfcni-conf"
-	testAPITimeout            = time.Second * 20
-	testRetryInterval         = time.Second * 1
+	testClusterName           = "dpu-operator-test-cluster"
+	testAPITimeout            = time.Second * 5
+	testRetryInterval         = time.Millisecond * 10
```
Comment (wizhaoredhat) [inline internal/controller/dpuoperatorconfig_controller_test.go:58]: On average, do you know how long it takes for an k8s API takes on kind? This seems a bit quick.

File: internal/controller/dpuoperatorconfig_controller_test.go [hunk 3]
```diff
@@ -60,193 +72,259 @@ func createDpuOperatorNameSpace() *corev1.Namespace {
...
+		LogLevel: 2,
+	}
+	return config
+}
+
+func createNameSpace(client client.Client, ns *v1.Namespace) {
+	// ignore error when createing the namespace since it can already exist
+	client.Create(context.Background(), ns)
+	found := v1.Namespace{}
+	err := client.Get(context.Background(), types.NamespacedName{Namespace: testNamespace, Name: ns.GetName()}, &found)
```
Comment (wizhaoredhat) [inline internal/controller/dpuoperatorconfig_controller_test.go:90]: We assume that namespaces are created instantly here. Would this always be the case? I expect similar code from `deleteNameSpace`

File: internal/controller/dpuoperatorconfig_controller_test.go [hunk 4]
```diff
@@ -9,18 +12,38 @@ import (
...
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/metrics/server"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	"k8s.io/client-go/tools/clientcmd"
+
+	configv1 "github.com/openshift/dpu-operator/api/v1"
+	//+kubebuilder:scaffold:imports
```
Comment (wizhaoredhat) [inline internal/controller/dpuoperatorconfig_controller_test.go:46]: Is this needed? What does it do for us.

File: internal/controller/dpuoperatorconfig_controller_test.go (hunk 2 above)
Comment (bn222) [reply internal/controller/dpuoperatorconfig_controller_test.go:58]: It takes <1 sec.

File: internal/controller/dpuoperatorconfig_controller_test.go (hunk 4 above)
Comment (bn222) [reply internal/controller/dpuoperatorconfig_controller_test.go]: Thanks for noticing. I've removed it.

File: internal/controller/dpuoperatorconfig_controller_test.go (hunk 3 above)
Comment (bn222) [reply internal/controller/dpuoperatorconfig_controller_test.go]: Good point

**Pull request merged**